lib/cocoapods-core/podfile/dsl.rb

```diff
@@ -273,6 +286,18 @@ def pod(name = nil, *requirements)
           raise StandardError, 'A dependency requires a name.'
         end
 
+        if requirements.empty? || requirements.last.is_a?(Hash)
+          theoretical_dependency = Dependency.new(name, requirements.last)
+          if theoretical_dependency.external_source.nil?
+            previously_defined_dependency = dependencies.find do |dependency|
```

Comment:
how will this linear scan perform on very large podfiled?

Reply:
There's a bit of a Shlemiel effect here, but ultimately it's all done in-memory and my profiling for 70 total pods shows an increase from 3% samples to 7%. 

What do you think? Should I optimize now or can this be done later? I mean, I've made several optimizations recently that outweigh this by a margin 😂

Reply:
Probably fine for now, but we might want to eventually put it in a hash for constant-time lookup